File: stages/neuralnetwork1.py
inputs = [1,2,3,2.5] #initialising inputs

# weights initialisation
weights = [[0.2,0.8,-0.5,1.0],
            [0.5, -0.91, 0.26, -0.5],
            [-0.26, -0.27, 0.17, 0.87]]

biases = [2,3,0.5]

layer_output = [] #output is a list

for neuron_weights, neuron_bias in zip(weights, biases):
    neuron_output = 0
    for n_input, weight in zip(inputs, neuron_weights):
        neuron_output += n_input*weight
    neuron_output += neuron_bias
    layer_output.append(neuron_output)

print(layer_output)

File: C#/Tuitorial-1/neuralNetwork1.cs
using System;
using System.Collections.Generic;
using System.Linq;

class NeuralNetwork
{
    static void Main(string[] args)
    {
        // Initialize inputs
        List<double> inputs = new List<double> { 1, 2, 3, 2.5 };

        // Initialize weights and biases
        List<List<double>> weights = new List<List<double>> {
            GenerateRandomWeights(4),
            GenerateRandomWeights(4),
            GenerateRandomWeights(4)
        };

        List<double> biases = new List<double> { 0, 0, 0 };

        // Calculate the output of each neuron
        List<double> layerOutput = weights.Select((neuronWeights, index) =>
            NeuronOutput(inputs, neuronWeights, biases[index])
        ).ToList();

        // Print the output of the layer
        Console.WriteLine(string.Join(", ", layerOutput.Select(o => Math.Round(o, 4))));
    }

    static double NeuronOutput(List<double> inputs, List<double> weights, double bias)
    {
        return inputs.Zip(weights, (input, weight) => input * weight).Sum() + bias;
    }

    static List<double> GenerateRandomWeights(int length)
    {
        Random random = new Random();
        return Enumerable.Range(0, length).Select(_ => random.NextDouble()).ToList();
    }
}
